File: agent-driven-architecture-main/examples/kotlin/src/test/fixtures/konsist/compliant/C17/blocks/triage/Project.kt
// COMPLIANT: MATCHING an Irreversible leaf stays legal everywhere — `is X ->` is a
// read, not a construction, which is the line C7's banner already draws.

package adr.blocks.triage

import adr.contract.Effect
import adr.contract.EscalationEffect

class Label {
    /**
     * Names PageOncall("nope") in PROSE, on a declaration's own KDoc, on purpose: a
     * rule that fired on a comment would be exactly the nuisance 15.2 warns about,
     * and `codeText` includes declaration KDoc. The rule matches the leaf's in-scope
     * CONSTRUCTION spellings, and a sentence is not one of them.
     */
    fun label(effect: Effect): String = when (effect) {
        is EscalationEffect.PageOncall -> "paged"
        else -> "other"
    }
}

File: agent-driven-architecture-main/examples/typescript/test/gate/fixtures/compliant/C17/src/blocks/console/project.ts
/**
 * COMPLIANT: MATCHING an Irreversible leaf stays legal everywhere — a closed
 * `switch`, a comparison and a type position are all reads, not constructions.
 *
 * This KDoc-shaped comment names PageOncall("nope") in PROSE on purpose: a rule
 * that fired on a declaration-attached comment would be exactly the nuisance
 * §15.2 warns about, and the check is written against the parse tree, so prose
 * cannot trip it.
 */
export function label(effect: { readonly kind: string }): string {
  switch (effect.kind) {
    case "PageOncall":
      return "paged";
    default:
      return effect.kind === "PageOncall" ? "paged" : "other";
  }
}
